the-collab-lab/tcl-70-smart-shopping-list
# Add ThemeContext and toggle theme functionality

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓  | :sparkles: New feature     |

## Acceptance Criteria

- [x] Toggle Control: A toggle switch in the NavBar.
- [x] Utilize Material-UI theming capabilities to ensure that components automatically adapt to the selected theme.

## Related Issue

Closes #51 

## Description 

This PR introduces a toggle for Dark and Light mode preferences using MUI, enhancing user experience by allowing users to choose their preferred visual mode. 

## Updates

### Before
<img width="1495" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/b9cc38cf-a7f3-4935-89a8-87eafcfc70f8">

### After
<img width="1495" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/42a8db97-f7fb-49a4-b1cb-d69954647c89">
<img width="1495" alt="image" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/121919044/984529e5-d41a-46fb-a3e1-7b599cf9c631">



## Testing Steps / QA Criteria

* Pull this branch with `git pull origin ag-dark-light-mode` and switch into it with `git checkout ag-dark-light-mode`.
* Find the sun/moon icon on the navbar and switch between dark and light modes. Observe theme change

## TODOs

- [ ] Design according to light/dark theme
